Apps/WebClient/src/Server/Controllers/UserProfileController.cs

```diff
@@ -0,0 +1,120 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Controllers
+{
+    using System.Diagnostics.Contracts;
+    using System.Security.Claims;
+    using System.Threading.Tasks;
+    using HealthGateway.Common.Authorization;
+    using HealthGateway.WebClient.Models;
+    using HealthGateway.WebClient.Services;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Web API to handle user profile interactions.
+    /// </summary>
+    [Authorize]
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/api/[controller]")]
+    [ApiController]
+    public class UserProfileController
+    {
+        private readonly IUserProfileService userProfileService;
+
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        private readonly IAuthorizationService authorizationService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserProfileController"/> class.
+        /// </summary>
+        /// <param name="userProfileService">The injected user profile service.</param>
+        /// <param name="authorizationService">The injected authorization service.</param>
+        /// <param name="httpContextAccessor">The injected http context accessor provider.</param>
+        public UserProfileController(
+            IUserProfileService userProfileService,
+            IHttpContextAccessor httpContextAccessor,
+            IAuthorizationService authorizationService)
+        {
+            this.userProfileService = userProfileService;
+            this.httpContextAccessor = httpContextAccessor;
+            this.authorizationService = authorizationService;
+        }
+
+        /// <summary>
+        /// Posts a user profile json to be inserted into the database.
+        /// </summary>
+        /// <returns>The http status.</returns>
+        /// <param name="userProfile">The user profile model.</param>
+        /// <response code="200">The user profile record was saved.</response>
+        /// <response code="400">The user profile was already inserted.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpPost]
+        [Authorize(Policy = "PatientOnly")]
+        public async Task<IActionResult> InsertUserProfile([FromBody] UserProfile userProfile)
```

Comment:
We should use PUT instead of POST since this URL will be inserting the record instead of updating.
What would be the url for this call? I believe it should be something like v1/.../UserProfile/{HDID}

Reply:
I think thats the other way around, PUT means update and POST means insert "Use PUT when you want to modify a singular resource which is already a part of resources collection. PUT replaces the resource in its entirety. Use PATCH if request updates part of the resource. Use POST when you want to add a child resource under resources collection."

Reply:
The URL is v1/api/UserProfile/
There's no need to pass the hdid since we already pass it in the body

Reply:
Take a look at this: https://restfulapi.net/rest-put-vs-post/  POST for create.

Reply:
Added the hdid to the endpoint, url should be v1/api/UserProfile/{hdid}